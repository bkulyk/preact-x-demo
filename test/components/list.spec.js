import React from 'react';
import render from 'preact-render-to-string';
import {
  describe,
  it,
  expect,
} from '../jest';
import List from '../../src/components/list';
import ListItem from '../../src/components/list-item';

jest.mock('../../src/components/list-item');
ListItem.mockImplementation(({ children }) => <span>{children}</span>);

describe('<List />', () => {
  it('should render a list with the items mocked out', async () => {
    const tree = render(<List />);
    expect(tree).toMatchSnapshot();
  });
});
