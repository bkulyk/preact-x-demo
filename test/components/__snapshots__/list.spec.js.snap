// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<List /> should do things 1`] = `"<ul __source=\\"[object Object]\\" __self=\\"[object Object]\\"><span __source=\\"[object Object]\\" __self=\\"[object Object]\\">a</span><span __source=\\"[object Object]\\" __self=\\"[object Object]\\">b</span><span __source=\\"[object Object]\\" __self=\\"[object Object]\\">c</span><span __source=\\"[object Object]\\" __self=\\"[object Object]\\">d</span></ul>"`;
